core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -3797,6 +3798,16 @@ public static List distinct(List list) {
     return new ArrayList<>(result);
   }
 
+  /** Support the MAP_KEYS function. */
+  public static List mapKeys(Map map) {
+    return (List) map.keySet().stream().collect(Collectors.toList());
```

Comment:
Why not use `new ArrayList<>(map.keySet())?`

Reply:
fixed @JiajunBernoulli